src/XMakeBuildEngine/BackEnd/Components/Logging/EventSourceSink.cs

```diff
@@ -129,6 +130,24 @@ public bool HaveLoggedBuildFinishedEvent
             get;
             set;
         }
+
+        /// <summary>
+        /// A list of warnings to treat as errors.  If null, nothing is treated as an error.  If an empty set, all warnings are treated as errors.
+        /// </summary>
+        public ISet<string> WarningsAsErrors
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// A list of build submission IDs that have logged errors.  If an error is logged outside of a submission, the submission ID is "0".
```

Comment:
I knew I left a comment with 0, this should say -1.  I thought the build submission IDs started at 1 when I coded this up but later figured out they start at zero.